core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandlerBase.java

```diff
@@ -153,10 +154,16 @@ protected CqlRequestHandlerBase(
               ? config.getDefaultProfile()
               : config.getProfile(profileName);
     }
-    this.queryPlan =
-        context
-            .getLoadBalancingPolicyWrapper()
-            .newQueryPlan(statement, executionProfile.getName(), session);
+    if (this.statement.getNode() != null) {
+      this.queryPlan = new ConcurrentLinkedQueue<Node>();
+      this.queryPlan.add(this.statement.getNode());
+
```

Comment:
Since JAVA-1883 there is a dedicated queue implementation for query plans, we can also use it here:
```java
this.queryPlan = new QueryPlan(this.statement.getNode());
```